eosmetrics/emtr-aggregate-timer.c

```diff
@@ -0,0 +1,163 @@
+{
+  EmtrAggregateTimer *self = user_data;
+  g_autofree gchar *timer_object_path = NULL;
+  g_autoptr(GError) error = NULL;
+
+  self->timer_proxy =
+    emer_aggregate_timer_proxy_new_for_bus_finish (result, &error);
+
+  if (error)
+    g_warning ("Error creating aggregate timer: %s", error->message);
+}
+
+static void
+on_server_aggregate_timer_started_cb (GObject      *source_object,
+                                      GAsyncResult *result,
+                                      gpointer      user_data)
+{
+  EmtrAggregateTimer *self = user_data;
+  g_autofree gchar *timer_object_path = NULL;
+  g_autoptr(GError) error = NULL;
+
+  emer_event_recorder_server_call_start_aggregate_timer_finish (self->dbus_proxy,
+                                                                &timer_object_path,
+                                                                result,
+                                                                &error);
+
+  if (error)
+    {
+      g_warning ("Error creating aggregate timer: %s", error->message);
+      return;
+    }
+
+
+  emer_aggregate_timer_proxy_new_for_bus (G_BUS_TYPE_SYSTEM,
+                                          G_DBUS_PROXY_FLAGS_NONE,
+                                          "com.endlessm.Metrics",
+                                          timer_object_path,
+                                          self->cancellable,
+                                          on_aggregate_timer_proxy_created_cb,
+                                          self);
+}
+
+EmtrAggregateTimer *
+emtr_aggregate_timer_new (EmerEventRecorderServer *dbus_proxy,
+                          GVariant                *event_id,
+                          GVariant                *aggregate_key,
+                          GVariant                *auxiliary_payload)
+{
+  EmtrAggregateTimer *self;
+
+  self = g_object_new (EMTR_TYPE_AGGREGATE_TIMER, NULL);
+  self->cancellable = g_cancellable_new ();
+  self->dbus_proxy = dbus_proxy;
+
+  emer_event_recorder_server_call_start_aggregate_timer (dbus_proxy,
+                                                         event_id,
+                                                         aggregate_key,
+                                                         TRUE,
+                                                         auxiliary_payload,
+                                                         self->cancellable,
+                                                         on_server_aggregate_timer_started_cb,
+                                                         self);
+
+  return self;
+}
+
+/**
+ * emtr_aggregate_timer_stop:
+ * @self: an #EmtrAggregateTimer
+ *
+ * Stops the aggregation timer. Timers must be stopped at most once.
+ */
+void
+emtr_aggregate_timer_stop (EmtrAggregateTimer *self)
+{
+  g_return_if_fail (EMTR_IS_AGGREGATE_TIMER (self));
+  g_return_if_fail (!self->stopped);
+
+  emer_aggregate_timer_call_stop_timer (self->timer_proxy, NULL, NULL, NULL);
```

Comment:
`self->timer_proxy` can be `NULL` at this point. A test case that I think would hit this:

```c
EmtrAggregateTimer *timer = emtr_aggregate_timer_new (...);
emtr_aggregate_timer_stop (timer);
```

Reply:
This should be covered now.